addons/hw_drivers/tools/upgrade.py

```diff
@@ -0,0 +1,209 @@
+
+
+def checkout(branch, remote=None, clean=True):
+    """Checkout to the given branch of the given git remote.
+
+    :param branch: The name of the branch to check out.
+    :param remote: The name of the local git remote to use (usually ``origin`` but computed if not provided).
+    :param clean: If True, clean the working directory after checking out.
+    """
+    _logger.info("Preparing local repository for checkout")
+    git('branch', '-m', branch)  # Rename the current branch to the target branch name
+
+    remote = remote or git('config', f'branch.{branch}.remote') or 'origin'
+    _ensure_production_remote(remote)
+
+    _logger.warning("Checking out %s/%s", remote, branch)
+    git('remote', 'set-branches', remote, branch)
+    git('fetch', remote, branch, '--depth=1', '--prune')  # refs/remotes to avoid 'unknown revision'
+    git('reset', f'{remote}/{branch}', '--hard')
+
+    if clean:
+        _logger.info("Cleaning the working directory")
+        git('clean', '-dfx')
+
+
+def update_requirements():
+    """Update the Python requirements of the IoT Box, installing the ones
+    listed in the requirements.txt file.
+    """
+    requirements_file = path_file('odoo', 'addons', 'iot_box_image', 'configuration', 'requirements.txt')
+    if not requirements_file.exists():
+        _logger.info("No requirements file found, not updating.")
+        return
+
+    _logger.warning("Updating pip requirements")
+    pip('install', '-r', requirements_file)
+
+
+@linux
+def update_packages():
+    """Update apt packages on the IoT Box, installing the ones listed in
+    the packages.txt file.
+    Requires ``writable`` context manager.
+    """
+    packages_file = path_file('odoo', 'addons', 'iot_box_image', 'configuration', 'packages.txt')
+    if not packages_file.exists():
+        _logger.info("No packages file found, not updating.")
+        return
+
+    # update and install packages in the foreground
+    commands = (
+        "export DEBIAN_FRONTEND=noninteractive && "
+        "mount -t proc proc /proc && "
+        "apt-get update && "
+        f"xargs apt-get -y -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' install < {packages_file}"
+    )
+    _logger.warning("Updating apt packages")
+    if subprocess.run(
+        f'sudo chroot /root_bypass_ramdisks /bin/bash -c "{commands}"', shell=True, check=False
+    ).returncode != 0:
+        _logger.error("An error occurred while trying to update the packages")
+        return
+
+    # upgrade and remove packages in the background
+    background_cmd = 'chroot /root_bypass_ramdisks /bin/bash -c "apt-get upgrade -y && apt-get -y autoremove"'
+    subprocess.Popen(["sudo", "bash", "-c", background_cmd], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+
+
+def misc_migration_updates():
+    """Run miscellaneous updates after the code update."""
+    _logger.warning("Running version migration updates")
+    if IS_LINUX:
```

Comment:
since only linux part is present it looks like this can be replaced with the same "linux" decorator as above